src/ProgressOnderwijsUtils/Data/ParameterizedSqlObjectMapper.cs

```diff
@@ -308,8 +307,39 @@ struct TRowReaderWithCols<T>
             static bool SupportsType([NotNull] Type type)
             {
                 var underlyingType = type.GetNonNullableUnderlyingType();
-                return getterMethodsByType.ContainsKey(underlyingType) ||
-                    isSqlDataReader && (underlyingType == typeof(TimeSpan) || underlyingType == typeof(DateTimeOffset));
+
+                return getterMethodsByType.ContainsKey(underlyingType)
+                    || isSqlDataReader && (underlyingType == typeof(TimeSpan) || underlyingType == typeof(DateTimeOffset))
+                    ;
+            }
+
+            static bool SupportsTypeByCustomCreateMethod([NotNull] Type type)
+            {
+                return CreateMethodOfTypeWithCreateMethod(type, out var _);
+            }
+
+            static bool CreateMethodOfTypeWithCreateMethod([NotNull] Type type, [CanBeNull] out MethodInfo methodInfo)
+            {
+                methodInfo = null;
+                var underlyingType = type.GetNonNullableUnderlyingType();
+                if (!underlyingType.IsValueType) {
+                    var method = underlyingType.GetMethod("Create", BindingFlags.Static | BindingFlags.Public);

```

Comment:
Ik zou het handiger vinden hier een iets minder algemene naam voor te kiezen; misschien "Materialize", "CreateFromData", "WrapPrimitiveType"; dwz, ik ben wat bang voor eventuele collisions met bestaande code (Create is iets wat ik iig een heel normale naam vind voor een static method die een factory is)

Reply:
Een alternatief is het markeren van de methode met een attribuut. Dan zijn we niet afhankelijk van een magische naam, maar is het expliciet gedefnieerd

Reply:
een abstract object werkt niet echt, omdat overerving in C# niet op klasseniveau maar op object niveau gaat, en je wilt hier een method op de class hebben. Je zou een interface kunnen gebruiken als 'marker' (a la IMetaObject). Op zich is een magische method wmb niet zo erg, alleen Create is een vrij normale naam die we nu ook al wel hebben, dus dat kent niet mijn voorkeur.

Reply:
IRL overleggen?  Wmb trouwens beter om deze PR niet specifiek op dit punt op te houden, want het is heel makkelijk aan te passen.